impl/src/main/java/org/jboss/cdi/tck/tests/build/compatible/extensions/syntheticObserverOfParameterizedType/SyntheticObserverOfParameterizedTypeExtension.java

```diff
@@ -0,0 +1,23 @@
+package org.jboss.cdi.tck.tests.build.compatible.extensions.syntheticObserverOfParameterizedType;
+
+import jakarta.enterprise.inject.build.compatible.spi.BuildCompatibleExtension;
+import jakarta.enterprise.inject.build.compatible.spi.Discovery;
+import jakarta.enterprise.inject.build.compatible.spi.ScannedClasses;
+import jakarta.enterprise.inject.build.compatible.spi.Synthesis;
+import jakarta.enterprise.inject.build.compatible.spi.SyntheticComponents;
+import jakarta.enterprise.inject.build.compatible.spi.Types;
+
+import java.util.List;
+
+public class SyntheticObserverOfParameterizedTypeExtension implements BuildCompatibleExtension {
+    @Discovery
+    public void discovery(ScannedClasses scan) {
+        scan.add(MyService.class.getName());
```

Comment:
@Ladicek This is redundant and causes failures for Weld.
`MyService` has `@ApplicationScoped` and as such will be discovered.
Will you fix that or should I?

Reply:
I would call that a bug _somewhere_. I intentionally added 1 or 2 cases where `@Discovery` is used to add a bean which is discovered through normal type discovery. I'm not aware of anything in the spec that would say this is illegal, so if that's what should happen in your opinion, we need to specify that somewhere.

Reply:
@Ladicek Yes you can do that. The discovered and added types (based off the same java class) are two different ones but with equal bean types and qualifiers (unless changed later on in the bootstrap OFC).
You should be getting an ambiguous resolution exception.

Taking portable extensions as an example - if you take a look at BBD javadoc, it has a specific parameter that serves this purpose - https://github.com/eclipse-ee4j/cdi/blob/master/api/src/main/java/jakarta/enterprise/inject/spi/BeforeBeanDiscovery.java#L162
In fact, you could in theory add one AT several times with just different IDs.

Reply:
My expectation actually was that the container would ignore the duplicate :-) We need to specify this somehow.

Reply:
This was specified in https://github.com/eclipse-ee4j/cdi/pull/571 (as non-portable) and the test was corrected, so resolving this open question.